target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -90,80 +104,115 @@ pub mod pyth_solana_receiver {
         Ok(())
     }
 
-    /// Post a price update using an encoded_vaa account and a MerklePriceUpdate calldata.
-    /// This should be called after the client has already verified the Vaa via the Wormhole contract.
-    /// Check out target_chains/solana/cli/src/main.rs for an example of how to do this.
-    #[allow(unused_variables)]
-    pub fn post_updates(ctx: Context<PostUpdates>, price_update: MerklePriceUpdate) -> Result<()> {
+
+    /// Post a price update using a VAA and a MerklePriceUpdate.
+    /// This function allows you to post a price update in a single transaction.
+    /// Compared to post_updates, it is less secure since you won't be able to verify all guardian signatures if you use this function because of transaction size limitations.
+    /// Typically, you can fit 5 guardian signatures in a transaction that uses this.
+    pub fn post_updates_atomic(
+        ctx: Context<PostUpdatesAtomic>,
+        params: PostUpdatesAtomicParams,
+    ) -> Result<()> {
+        // This section is borrowed from https://github.com/wormhole-foundation/wormhole/blob/wen/solana-rewrite/solana/programs/core-bridge/src/processor/parse_and_verify_vaa/verify_encoded_vaa_v1.rs#L59
+        let vaa = Vaa::parse(&params.vaa).map_err(|_| ReceiverError::DeserializeVaaFailed)?;
+        // Must be V1.
+        require_eq!(vaa.version(), 1, ReceiverError::InvalidVaaVersion);
+
+        // Make sure the encoded guardian set index agrees with the guardian set account's index.
+        let guardian_set = ctx.accounts.guardian_set.inner();
+        require_eq!(
+            vaa.guardian_set_index(),
+            guardian_set.index,
+            ReceiverError::GuardianSetMismatch
+        );
+
+        // Do we have enough signatures for quorum?
```

Comment:
comment seems wrong (this doesn't seem to check anything)

Reply:
maybe leave a comment here that we don't need to check the number of signatures, but users of posted price updates are *required* to check the `verified_signatures` field themselves.

Reply:
by the way, asking pyth users to check `verified_signatures` is somewhat scary. It's the kind of check people can easily forget about. we should think about how to make this less error prone.

Reply:
I have an idea about this